packages/bruno-electron/src/app/first-launch.js

```
@@ -0,0 +1,120 @@
+const fs = require('fs');
+const fsPromises = require('fs').promises;
+const path = require('path');
+const { app } = require('electron');
+const { preferencesUtil } = require('../store/preferences');
+const { importCollection } = require('../utils/collection-import');
+
+/**
+ * Handles first launch logic including sample collection import
+ */
+class FirstLaunchHandler {
+  constructor(mainWindow, collectionWatcher, lastOpenedCollections) {
+    this.mainWindow = mainWindow;
+    this.collectionWatcher = collectionWatcher;
+    this.lastOpenedCollections = lastOpenedCollections;
+  }
+
+  /**
+   * Check if this is the first launch and handle sample collection import
+   */
+  async handleFirstLaunch() {
+    try {
+      // Check if this is the first launch
+      if (!preferencesUtil.isFirstLaunch()) {
+        console.log('FirstLaunchHandler: Not first launch, exiting');
+        return;
+      }
+
+      // Get the default collection location
+      const collectionLocation = await this.getDefaultCollectionLocation();
+
+      // Import the sample collection
+      await this.importSampleCollection(collectionLocation);
+
+      preferencesUtil.markAsLaunched();
+    } catch (error) {
+      console.error('Failed to handle first launch:', error);
+      // Still mark as launched to prevent retry on next startup
+      preferencesUtil.markAsLaunched();
+    }
+  }
+
+  /**
+   * Get the default location for collections
+   */
+  async getDefaultCollectionLocation() {
```

@sanjaikumar-bruno we will have to check if  we really need the if conditions. It is highly unlikely that electron will return a path that doesn't exist. It would rather throw an error it the app.getPath cannot find the concerned path. will you be able to check the exact behaviour on how app.getPath works and  write the function according to it.

But with this logic, desktop will never get used if app.getPath throws an error.

---

Updated the logic to catch the error.